src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConverterToolViewModel.cs

```diff
@@ -0,0 +1,249 @@
+                    }
+                    else
+                    {
+                        HasInvalidFilesSelected = true;
+                    }
+                }
+            }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region SelectFilesBrowseCommand
+
+        public IAsyncRelayCommand SelectFilesBrowseCommand { get; }
+
+        private async Task ExecuteSelectFilesBrowseCommandAsync()
+        {
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                HasInvalidFilesSelected = false;
+
+                var filePicker = new FileOpenPicker
+                {
+                    ViewMode = PickerViewMode.List,
+                    SuggestedStartLocation = PickerLocationId.ComputerFolder
+                };
+
+                for (int i = 0; i < SupportedFileExtensions.Length; i++)
+                {
+                    filePicker.FileTypeFilter.Add(SupportedFileExtensions[i]);
+                }
+
+                IReadOnlyList<StorageFile> files = await filePicker.PickMultipleFilesAsync();
+                for (int i = 0; i < files.Count; i++)
+                {
+                    QueueNewConversion(files[i]);
+                }
+            });
+        }
+
+        #endregion
+
+        #region DeleteAllCommand
+
+        public IRelayCommand DeleteAllCommand { get; }
+
+        private void ExecuteDeleteAllCommand()
+        {
+            ConversionWorkQueue.Clear();
+        }
+
+        #endregion
+
+        #region SaveAllCommand
+
+        public IAsyncRelayCommand SaveAllCommand { get; }
+
+        private async Task ExecuteSaveAllCommandAsync()
+        {
+            var works = ConversionWorkQueue.ToList();
+
+            var folderPicker = new FolderPicker
+            {
+                ViewMode = PickerViewMode.List
+            };
+
+            StorageFolder? selectedFolder = await folderPicker.PickSingleFolderAsync();
+
+            if (selectedFolder is not null)
+            {
+                foreach (ImageConversionWorkItem work in works)
+                {
+                    StorageFile newFile = await selectedFolder.CreateFileAsync(work.FileName, CreationCollisionOption.ReplaceExisting);
+                    await newFile.RenameAsync(string.Concat(newFile.DisplayName, ImageHelper.GetExtension(ConvertedFormat)), NameCollisionOption.GenerateUniqueName);
+                    using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+                    {
+                        BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(ConvertedFormat, outputStream);
+                        encoder.SetSoftwareBitmap(work.Bitmap);
+                        await encoder.FlushAsync();
+                    }
+
+                    work.DeleteCommand.Execute(null);
+                }
+            }
+        }
+
+        #endregion
+
+        private void QueueNewConversion(StorageFile file)
```

Comment:
Since you're doing the conversion of image when the user clicks on Save, I'd argue that either this method should renamed, because we're technically not queuing a conversion yet, or that the UI should be revamped.
It's a little confusing to me that we should "save" in order to "perform a conversion".

Reply:
I have renamed the method to ```QueueNewConversionWorkItem```. Please, let me know if you think another name would  fit better.

Reply:
`QueueNewConversionWorkItem` works, yes. :)